File: 1365.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {

        Map<Integer, Integer> hashMap = new HashMap<>();
        int[] res = Arrays.copyOf(nums, nums.length);
        Arrays.sort(res);
        
        for (int i = 0; i < nums.length; i++) {
            if (!hashMap.containsKey(res[i])) {
                hashMap.put(res[i], i);
            }
        }

        for (int i = 0; i < nums.length; i++) {
            res[i] = hashMap.get(nums[i]);
        }

        return res;
    }
}

File: 1365.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        hashMap = {}

        res = sorted(nums[:])

        for i, num in enumerate(res):
            if num not in hashMap:
                hashMap[num] = i 

        for i, num in enumerate(nums):
            res[i] = hashMap[num]

        return res
